frontend/src/containers/pages/AuthoringTool/CanvasSideBar/SceneSettings.jsx

```diff
@@ -51,6 +54,25 @@ export default function SceneSettings() {
     (checked) => checked
   );
 
+  const { scenarioId } = useParams();
+  const { scenes } = useContext(SceneContext);
+  const { getUserIdToken } = useContext(AuthenticationContext);
+  async function saveRoles(newRoles) {
+    const updatedScenes = scenes.map(({ _id, name, roles: oldRoles }) => {
+      const roles = newRoles || oldRoles;
```

Comment:
this implicitly updates **every** scene which is unexpected behaviour.

We would expect it to only update for the current scene we are on.

Reply:
fixed!

Reply:
also fixed UI initialization issue, the saved choice should now appear